state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,329 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+IMU Sensor Test
+* Author(s): Yashika Batra
+"""
+
+import time
+from ulab import numpy
+from lib import pycubed as cubesat
+
+
+wait_time = 3
+norm = numpy.linalg.norm
+
+
+def request_imu_data(prompt):
+    """
+    Ask the user to type y/any key to start/cancel the test
+    If y, gather IMU data
+    Accelerometer and Gyroscope readings are collected 10 times
+    in wait_time seconds
+    Magnetometer readings are collected at the start and end
+    of wait_time seconds
+    If any other key, return None for all readings
+    """
+    # ask user whether or not to start the test
+    print(prompt, end="")
+    start_test = input("Type Y to start the test, any key to cancel: ")
+
+    # if user opts to cancel the test, return None, handle in the next function
+    if start_test.lower() != "y":
+        return None, None, None
+
+    # else proceed with the test
+    time.sleep(0.5)
+    print("Collecting IMU data...")
+
+    # collect magnetometer reading at the start of wait time
+    start_mag = numpy.array(cubesat.magnetic())
+
+    # collect total accelerometer and gyroscope readings
+    # do 10 reads over wait_time
+    acc_total = numpy.array([0., 0., 0.])
+    gyro_total = numpy.array([0., 0., 0.])
+    for i in range(10):
+        acc_total += numpy.array(cubesat.acceleration())
+        gyro_total += numpy.array(cubesat.gyro())
+        time.sleep(wait_time / 10)
+
+    # collect magnetometer reading at the end of wait time
+    end_mag = numpy.array(cubesat.magnetic())
+
+    # calculate the average acc and gyro readings
+    acc_final = acc_total / 10
+    gyro_final = gyro_total / 10
+
+    # calculate the change in the total magnetometer reading
+    mag = norm(end_mag - start_mag)
+
+    # return acc, gyro, change in mag
+    print("Data Collection Complete")
+    return acc_final, gyro_final, mag
+
+
+def check_gravity_acc(acc, direction):
+    if direction.lower() == "x":
```

Comment:
I think a better way to do this is to check if acceleration is about 9.8 `abs(norm(acc)-9.8 <= 1`. And to check if it's roughly parallel (or antiparallel) to the vectors `norm(dot(acc, [1,0,0]))>=0.9`